### DISCUSSION TITLE
How to join a private/invisible room

### DISCUSSION THREAD

**Discussion by guguxh** [wxyz-abcd/node-haxball#46] [General]
I want to do this in your website, but I don't know how to.

**Comment by wxyz-abcd** [comment]
We can only join private room if we know its id. We can do this by appending original haxball website's query parameters into our website's url. For example, if the room link is https://www.haxball.com/play?c=31IBNI3w4F0, you can join this room using this url: https://abc-haxball-proxy.up.railway.app/?no_proxy_server=true&c=31IBNI3w4F0
If you have a password-protected room, you can also use pwd parameter like this for auto-join: https://abc-haxball-proxy.up.railway.app/?no_proxy_server=true&c=31IBNI3w4F0&pwd=123
If you are trying to join a recaptcha-protected room, you can write the recaptcha token this way: https://abc-haxball-proxy.up.railway.app/?no_proxy_server=true&c=31IBNI3w4F0&jtkn=....

There are also parameters to automate creating a room:

name: the room's name
pwd: the room's password
mp: max player count
ctkn: token for creating the room
csil: show in room list. (0 or 1)
